app/api.py

```diff
@@ -0,0 +1,195 @@
+        logger.info(f"Received uploaded file and saved to {file_location}")
+
+        # Extract text and boxes from PDF
+        _, text_blocks = extract_text_and_boxes_from_pdf(file_location)
+
+        # Store the extracted text in the vector database
+        for i, text in enumerate(text_blocks):
+            doc_id = f"{file.filename}_page_{i+1}"
+            store_text_in_vector_db(doc_id, text)
+            logger.info(f"Stored text from page {i+1} in vector database.")
+
+        return {"message": "PDF processed and text stored in vector database."}
+    except Exception as e:
+        logger.error(f"Error processing PDF: {e}")
+        raise HTTPException(status_code=500, detail=str(e))
+    finally:
+        # Clean up the saved file
+        if file_location and Path(file_location).exists():
+            Path(file_location).unlink(missing_ok=True)
+
+
+@app.post("/extract/")
+async def extract_names_endpoint(name_pairs: NamePairs = Depends(get_name_pairs)):
+    try:
+        logger.info(f"Received name_pairs: {name_pairs}")
+
+        file_location = name_pairs.file_location
+        name_list = name_pairs.names
+
+        if not file_location:
+            raise HTTPException(
+                status_code=400, detail="File location must be provided"
+            )
+
+        # Assuming extract_names_and_boxes and extract_text_and_boxes_from_pdf are defined
+        if file_location.lower().endswith((".png", ".jpg", ".jpeg")):
+            logger.info(f"Processing image file: {file_location}")
+            extracted_data, _ = extract_names_and_boxes(
+                file_location, det_model, reco_model
+            )
+        elif file_location.lower().endswith(".pdf"):
+            logger.info(f"Processing PDF file: {file_location}")
+            extracted_data, _ = extract_text_and_boxes_from_pdf(file_location)
+        else:
+            raise HTTPException(status_code=400, detail="Unsupported file type")
+
+        logger.info(f"Extracted data: {extracted_data}")
+
+        matched_names = []
+        for full_name in name_list:
+            matches = process.extract(
+                full_name, [d["name"] for d in extracted_data], scorer=fuzz.ratio
+            )
+            logger.info(f"Matches for '{full_name}': {matches}")
+            for match in matches:
+                if match[1] >= 90:
+                    for data in extracted_data:
+                        if data["name"] == match[0]:
+                            matched_names.append(
+                                {
+                                    "provided_name": full_name,
+                                    "extracted_name": data["name"],
+                                    "bbox": data["bbox"],
+                                    "score": match[1],
+                                }
+                            )
+
+        logger.info(f"Matched Names: {matched_names}")
+        return JSONResponse(content={"matched_names": matched_names})
```

Comment:
Define a return schema with pydantic and return instead the instance of that class. That will help you with typing check and will save you a lot of issues when modifying the endpoint and with the itnegrations. https://fastapi.tiangolo.com/tutorial/response-model/